File: java/src/yax/test/KeywordMap.java
/* Copyright 2009, UCAR/Unidata and OPeNDAP, Inc.
   See the COPYRIGHT file for more information. */

package yax.test;

import java.util.HashMap;
import java.util.Map;

public class KeywordMap
{
    static public class Keyword
    {
        String name;
        int opentag;
        int closetag;
        int attrtag;
        public Keyword(String name, int opentag, int closetag, int attrtag)
        {
            this.name = name;
            this.opentag = opentag;
            this.closetag = closetag;
            this.attrtag = attrtag;
        }
    };
    
    Map<String,Keyword> keywords;
    
    public KeywordMap()
    {
        keywords = new HashMap<String,Keyword>();
    }
    
    public void add(Keyword keyword)
    {
	keywords.put(keyword.name,keyword);
    }

    public void add(String name, int opentag, int closetag, int attrtag)
    {
	add(new Keyword(name,opentag,closetag,attrtag));
    }

    public Keyword get(String name)
    {
	if(name == null) return null;
	return keywords.get(name);
    }
    

} // class KeywordMap    
    

File: python/yax/keywordmap.py
# Copyright 2009, UCAR/Unidata
#   See the COPYRIGHT file for more information.

# Probably can get rid of this class
class KeywordMap:

  def __init__() :
    keywords = {}

  def addKeyword(keyword):
    keywords += {keyword.name:keyword}

  def add(name, opentag, closetag, attrtag) :
    addKeyword(Keyword(name,opentag,closetag,attrtag))

  def get(name):
    if(name == None) : return None
    return keywords[name]
# end class KeywordMap

class Keyword:
  def __init__(name, opentag, closetag, attrtag) :
    self.name = name
    self.opentag = opentag
    self.closetag = closetag
    self.attrtag = attrtag

#end class Keyword
